completions/make.lua

```diff
@@ -0,0 +1,145 @@
+local clink_version = require('clink_version')
+if not clink_version.supports_argmatcher_delayinit then
+    print("make.lua argmatcher requires a newer version of Clink; please upgrade.")
+    return
+end
+
+-- Table of special targets to always ignore.
+local special_targets = {
+    ['.PHONY'] = true,
+    ['.SUFFIXES'] = true,
+    ['.DEFAULT'] = true,
+    ['.PRECIOUS'] = true,
+    ['.INTERMEDIATE'] = true,
+    ['.SECONDARY'] = true,
+    ['.SECONDEXPANSION'] = true,
+    ['.DELETE_ON_ERROR'] = true,
+    ['.IGNORE'] = true,
+    ['.LOW_RESOLUTION_TIME'] = true,
+    ['.SILENT'] = true,
+    ['.EXPORT_ALL_VARIABLES'] = true,
+    ['.NOTPARALLEL'] = true,
+    ['.ONESHELL'] = true,
+    ['.POSIX'] = true,
+    ['.NOEXPORT'] = true,
+    ['.MAKE'] = true,
+}
+
+local function extract_target(line, last_line, targets, allow_pathlike)
```

Comment:
Can you add (back) a comment describing the `extract_target` function?

I would like for diffing nmake.lua vs make.lua to show necessary functional changes.  There are a lot of whitespace changes, removed comments, and some renamed variables.  After this PR is merged, I'll resolve the whitespace changes and renamed variables so they're consistent between the two files.